### ISSUE TITLE
&<modifierClass> will count as a deeper nesting level

### ISSUE THREAD

**Issue by andreechristian** [brigade/scss-lint#829]
Hello,

I am using scss lint in Atom, and i specify the rules of nesting depth should be no greater than 3. I have an edge case where i need to specify something like this

```
.banner {
  <some rules>

  &-navigation {
    <some rules>

    a {
       color: $c-red;

       &.is-active {
         color: $c-green;
       }
    }
  }
}
```

i believe that the output should be equal to

```
.banner {
  <some rules>
}

.banner-navigation {
  <some rules>
}

.banner-navigation a {
  color: #f00;
}

.banner-navigation a.is-active{
  color: #0f0;
}
```

which shouldn't break nesting depth rules that i specified.
yet i get this warning `Nesting should be no greater than 3, but was 4` at `&.is-active {` line

is it a bug that scss lint count `&` as a deeper level?

**Comment by sds** [comment]
Hey @andreechristian,

`NestingDepth` is working as designed here. See https://github.com/brigade/scss-lint/issues/748#issuecomment-192398322:

> The linter is working as expected here. `NestingDepth` doesn't refer to selector depth (that's what the `SelectorDepth` linter is for), it refers to the literal syntactic nesting you can do with Sass. In your case, your `&:hover` _is_ at depth 4.
> 
> The rationale here is that the more nested your code is the harder it is to understand. Your example is a simple one, but it's quite easy to find examples where you are deep in a nested block and have to scroll-up to figure out what the parent reference `&` is actually referring to.
> 
> If you don't like the warnings in this case, disable it for the file in question, or disable the linter entirely.
> 
> Hope that helps.
